dali/kernels/common/join/tensor_join_cpu.h

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_COMMON_JOIN_TENSOR_JOIN_CPU_H_
+#define DALI_KERNELS_COMMON_JOIN_TENSOR_JOIN_CPU_H_
+
+#include <vector>
+#include "dali/kernels/kernel.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/core/format.h"
+
+namespace dali {
+namespace kernels {
+namespace impl {
+
+template<typename T>
+void
+ConcatenateTensors(TensorView<StorageCPU, T> output, span<TensorView<StorageCPU, const T>> inputs,
+                   int axis) {
+  SmallVector<int64_t, 8> copy_sizes;
```

Comment:
Does it make sense to check if `inputs[t].shape.begin() + axis < inputs[t].shape.end()`?

Reply:
This is an internal function (see `namespace impl`), so the correctness of input is assumed - that's how I understand it, at least.

Reply:
That's right, input correctness is assumed for the functions in `impl`. It's checked in `Setup` and `Run`